dd-sdk-android-ndk/src/main/cpp/utils/signal-monitor.c

```diff
@@ -0,0 +1,193 @@
+#include "signal-monitor.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <jni.h>
+#include <pthread.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "../datadog-native-lib.h"
+
+#define HANDLED_SIGNAL_COUNT 6
+
+void handle_signal(int signum, siginfo_t *info,
+                   void *user_context);
+
+bool configure_signal_stack(void);
+
+stack_t signal_stack;
+/**
+ * Global shared context reports
+ */
+static JNIEnv *global_env;
+/* the signal handler array */
+struct sigaction *global_sigaction;
+
+/* the previous signal handler array */
+struct sigaction *global_previous_sigaction;
+
+/**
+ * Native signals which will be captured by the signal handler™
+ */
+
+struct signal {
+    int signal_value;
+    char *signal_name;
+    char *signal_error_message;
+};
+
+// TODO https://datadoghq.atlassian.net/browse/RUMM-576
+// We should use a Hashtable (Hashmap) here. For now I could not find something in C so probably
+// will have to implement one or just use a pure array[key] -> value implementation even though
+// this will take more memory.
+static const struct signal handled_signals[HANDLED_SIGNAL_COUNT] = {
+        {SIGILL,  "SIGILL",  "Illegal instruction"},
+        {SIGTRAP, "SIGTRAP", "Trace/breakpoint trap"},
+        {SIGABRT, "SIGABRT", "Abort program"},
+        {SIGBUS,  "SIGBUS",  "Floating-point exception"},
+        {SIGFPE,  "SIGFPE",  "Bus error (bad memory access)"},
+        {SIGSEGV, "SIGSEGV", "Segmentation violation (invalid memory reference)"}
+};
+
+static pthread_mutex_t handlerMutex = PTHREAD_MUTEX_INITIALIZER;
+
+size_t handled_signals_size() { return sizeof(handled_signals) / sizeof(handled_signals[0]); }
```

Comment:
If you have this function, why do you need `HANDLED_SIGNAL_COUNT`?

Reply:
was just for initialisation...legacy code...but I will remove it you are right.